### ISSUE TITLE
Request: Tie in to Apple HomeKit

### ISSUE THREAD

**Issue by stevenmcastano** [bwssytems/ha-bridge#159]
First of all, thanks for this... this is EXACTLY what I've been looking for. I've been writing my own custom Alexa skill in python for a month, and this replaces the entire thing with even more control and better functionality.

Do you think there's any chances of this picking up the ability to tie into Apple HomeKit and something like the MyTouchHome app?

Right now I'm using another project called "homebridge" [https://github.com/nfarina/homebridge](https://github.com/nfarina/homebridge) to link my iOS devices to my multiple home automation systems for the same functionality. So I could just configure both of them with matching names and activation URLs, etc, but it would be amazing to have both of these in one single app.

Any plans for that in the future?

**Comment by bwssytems** [comment]
Something to be looked into.

**Comment by stevenmcastano** [comment]
For sure, I'm using it more and more... I've been tying in shell scripts, multiple command options, etc... and things are going great. The only problem is, right now I have to replicate all those changes over to another app for Siri to work as well.

Is there an API in ha-bridge I could hit? This way maybe my homebridge install could at least just point to ha-bridge and I'd only have to send a single web request over and use ha-bridge to do all the actual heavy lifting?

**Comment by audiofreak9** [comment]
Homekit is Phillips Hue Bridge compatible, it should discover the ha-bridge as a Hue if you run the ha-bridge on port 80 rather than the default port 8080.

**Comment by bwssytems** [comment]
Also, the ha-bridge emulates a Philips Hue Bridge so you can use those api calls to control it as well. ans as @audiofreak9 says, most external programs that utilize a hue, it is better to set the ha-brige to port 80 as a lot of apps hard code this port number for a hue bridge

**Comment by magenbs** [comment]
Homekit only works with the new Hue bridge 2.  Does ha-bridge emulate the new authentication?

http://www2.meethue.com/en-us/friends-of-hue/apple-homekit/

**Comment by bwssytems** [comment]
Yes, I incorporated the features for V2 into the V3.0.0 and later ha-bridge. There may be some nuances of V2 that I am missing.

**Comment by Ukhauk** [comment]
Siri is a great addition to Alexa integration. To add to ha-bridge into homekit what Apps are best to pick up the bridge and devices ? I have been trying the Philips Hue apps which pick up ha-bridge, they want me to update the Hue Bridge software, if I skip this and go to Siri Voice control in the app it tries to pair my phone with the bridge but says device not found etc. So I am guessing I need a better 3rd party app thats not as fussy. Much appreciated

**Comment by bwssytems** [comment]
Which app are you using? I'll give it a try and see if I can do some debug.

**Comment by Ukhauk** [comment]
Both versions of the UK Philips Hue app. Hesperus, Eve and MyHome+. The Hue apps see ha_bridge and conplain that the firmaware needs updating. What apps are you using to add ha_bridge to homekit ?

**Comment by bwssytems** [comment]
I do not use Homekit, not an apple person. Just was looking for what apps do what to look at.

**Comment by bwssytems** [comment]
I have an updated jar to try: http://bwssystems.com/files/ha-bridge-3.1.0l.jar

**Comment by rodriguezst** [comment]
It still doesn't work with Apple Homekit for me but with the updated jar I got rid of those annoying messages prompting to update the bridge firmware version while using the official Hue app (v1) for Android /iOS.
Thanks!

As for the Hue for Android/iOS (v2) app, it detects the bridge just fine but then prompts you to define new rooms that cannot be created. I guess your project does not support this feature.

**Comment by peadard** [comment]
Ok so bear with me for a tick. I have the same issue so I looked into an alternative way of getting this to work with HomeKit. Having this Hue Bridge setup I then setup the nfarina homebridge [https://github.com/nfarina/homebridge](url). Once I had this up and running I installed the Hue accessory here: [https://www.npmjs.com/package/homebridge-philipshue](url).
I then pointed the accessory to my bwssystems hue bridge and voila it discovered the bridge and all its accessories. Now while it is not a perfect way of doing it, it does mean that I only have to add my accessories in one place (bwssystems) and Alexa and Siri can both access them. I hope this made sense to you guys.

**Comment by rodriguezst** [comment]
Thanks @peadard!

I don't plan on using Homekit that much (more of an Android guy) but I'll do it your way to play a bit with Apple's Homekit

**Comment by bwssytems** [comment]
V3.2.0 has been release, see if that helps with home kit.

**Comment by rodriguezst** [comment]
V3.2.0 still doesn't work with HomeKit
After reading a bit about how HomeKit works I found out that it doesn't use the public Hue api. 
Hue bridges with HomeKit support have another Apple's custom protocol implemented with a setup pin and Apple's Zeroconf (Bonjour).

I ended up installing nfarina's homebridge as @peadard suggested and everything works great. All my ha-bridge accessories are now available on Apple devices.
